core/src/main/java/io/questdb/griffin/SqlUtil.java

```diff
@@ -321,7 +321,19 @@ public static byte implicitCastDoubleAsByte(double value) {
             return 0;
         }
 
-        throw ImplicitCastException.inconvertibleValue(value, ColumnType.FLOAT, ColumnType.BYTE);
+        throw ImplicitCastException.inconvertibleValue(value, ColumnType.DOUBLE, ColumnType.BYTE);
+    }
+
+    public static byte implicitCastDoubleAsByte(double value, int fromType) {
+        if (value >= Byte.MIN_VALUE && value <= Byte.MAX_VALUE) {
+            return (byte) value;
```

Comment:
This returns the incorrect result; should be an empty data set, but it returns the same rows as `= 1.0`

```Java
    @Test
    public void testIntToByteCast() throws Exception {
        assertQuery("id\tstatus\n",
                "select id, case when status = 1.5 then 'active' end as status from foo;",
                "create table foo as (select x as id, rnd_byte(0, 1) as status from long_sequence(5));",
                null,
                true,
                true);
    }
```